internal/supervisor/installation.go

```diff
@@ -540,46 +519,109 @@ func (s *InstallationSupervisor) updateInstallation(installation *model.Installa
 		}
 	}
 
-	logger.Infof("Finished updating clusters installations")
+	logger.Info("Finished updating clusters installations")
 
-	return model.InstallationStateUpdateInProgress
+	return s.waitForUpdateStable(installation, instanceID, logger)
 }
 
-func (s *InstallationSupervisor) waitForUpdateComplete(installation *model.Installation, instanceID string, logger log.FieldLogger) string {
+func (s *InstallationSupervisor) waitForUpdateStable(installation *model.Installation, instanceID string, logger log.FieldLogger) string {
+	stable, err := s.checkIfClusterInstallationsAreStable(installation, logger)
+	if err != nil {
+		logger.WithError(err).Error("Installation update failed")
+		return model.InstallationStateUpdateFailed
+	}
+	if !stable {
+		return model.InstallationStateUpdateInProgress
+	}
+
+	logger.Info("Finished updating installation")
+
+	return model.InstallationStateStable
+}
+
+func (s *InstallationSupervisor) hibernateInstallation(installation *model.Installation, instanceID string, logger log.FieldLogger) string {
 	clusterInstallations, err := s.store.GetClusterInstallations(&model.ClusterInstallationFilter{
-		InstallationID: installation.ID,
 		PerPage:        model.AllPerPage,
+		InstallationID: installation.ID,
 	})
 	if err != nil {
 		logger.WithError(err).Warn("Failed to find cluster installations")
 		return installation.State
 	}
 
-	var stable, reconciling, failed int
+	var clusterInstallationIDs []string
+	if len(clusterInstallations) > 0 {
```

Comment:
nit: could we invert this `if` and `return` early so we could outdent the inner block? I see you'd have to repeat the `return` statement but imho that's worth it. 0/5 though.